app/v2/connectors/AuditingConnector.scala

```diff
@@ -50,14 +53,17 @@ class AuditingConnectorImpl @Inject() (httpClient: HttpClientV2, appConfig: AppC
     with HasMetrics
     with Logging {
 
-  def post(auditType: AuditType, source: Source[ByteString, _], contentType: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Unit] =
+  def post(auditType: AuditType, source: Source[ByteString, _], contentType: String, contentLength: Long)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[Unit] =
     withMetricsTimerAsync(MetricsKeys.AuditingBackend.Post) {
       _ =>
         val url = appConfig.auditingUrl.withPath(auditingRoute(auditType))
 
         httpClient
           .post(url"$url")
-          .transform(_.addHttpHeaders(HeaderNames.CONTENT_TYPE -> contentType))
+          .transform(_.addHttpHeaders(HeaderNames.CONTENT_TYPE -> contentType, "ContentLength" -> contentLength.toString))
```

Comment:
If we're going to use a header for the length, as I said in the other PR, it needs to start with `X-`

Reply:
Yep.

Reply:
Yep.